File: Fabric/src/test/java/team/zeds/testjava/MyMultiblocks.java
package team.zeds.testjava;

import kotlin.Unit;
import team.zeds.structured.MultiblockBuilder;
import team.zeds.structured.MultiblockRegistries;
import team.zeds.structured.settings.MultiblockStructure;

public class MyMultiblocks {
    public static final MultiblockStructure MY_FIRST_MULTIBLOCK;

    static {
        MY_FIRST_MULTIBLOCK = MultiblockRegistries.registerMultiblock(
                new MultiblockBuilder()
                        .configure(config -> {
                            config.setModelName("my_first_multiblock");
                            config.setOnOpen((player, pos) -> {
                                //My code
                                return Unit.INSTANCE;
                            });
                            return Unit.INSTANCE;
                        })
                        .build()
        );
    }

    public static void init() {}
}

File: Fabric/src/test/java/team/zeds/testkotlin/MyMultiblocks.kt
package team.zeds.testkotlin

import net.minecraft.core.BlockPos
import net.minecraft.world.entity.player.Player
import team.zeds.structured.MultiblockBuilder
import team.zeds.structured.MultiblockRegistries.registerMultiblock
import team.zeds.structured.settings.MultiblockStructure

object MyMultiblocks {
    val myFirstMultiblock: MultiblockStructure

    init {
        myFirstMultiblock = registerMultiblock(
            MultiblockBuilder()
                .configure {
                    modelName = "my_first_multiblock"
                    onOpen = { player: Player, pos: BlockPos ->
                        //there your code
                    }
                }
                .build()
        )
    }
    @JvmStatic
    fun init() {}
}